manager/src/main/java/io/gingersnapproject/configuration/Configuration.java

```diff
@@ -2,13 +2,36 @@
 
 import java.util.Map;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.quarkus.logging.Log;
 import io.smallrye.config.ConfigMapping;
 import io.smallrye.config.WithName;
 
 @ConfigMapping(prefix = "gingersnap")
 public interface Configuration {
+   static final Logger log = LoggerFactory.getLogger(Configuration.class);
+
    HotRod hotrod();
 
    @WithName("rule")
    Map<String, Rule> rules();
+   
+   default void addRule(String name, Rule rule) {
+      var ruleInMap = rules().putIfAbsent(name, rule);
+      if ( ruleInMap != null) {
+         throw new UnsupportedOperationException("Existing Rule("+name+") can't be updated.");
+      }
+      // TODO: add the rule to the application
```

Comment:
I think we should create a separate interface and implementation to handle runtime rules. That way we can utilise `javax.enterprise.event.Event`s to handle additions/removals in the various Beans, i.e. `DatabaseHandler`, `IndexingHandler` and `Caches, by simply observing these events.

The db-syncer `NotificationManager` takes a similar approach for various lifecycle events.